test/schemes/fluid/surface_normal_sph.jl

```diff
@@ -0,0 +1,269 @@
+    return boundary_system
+end
+
+function create_fluid_system(coordinates, velocity, mass, density, particle_spacing;
+                             NDIMS=2, smoothing_length=1.0, wall=false, walldistance=0.0)
+    tspan = (0.0, 0.01)
+
+    fluid = InitialCondition(coordinates=coordinates,
+                             velocity=velocity,
+                             mass=mass,
+                             density=density,
+                             particle_spacing=particle_spacing)
+
+    state_equation = StateEquationCole(sound_speed=10.0,
+                                       reference_density=1000.0,
+                                       exponent=1)
+
+    kernel = SchoenbergCubicSplineKernel{NDIMS}()
+
+    system = WeaklyCompressibleSPHSystem(fluid,
+                                         SummationDensity(),
+                                         state_equation,
+                                         kernel,
+                                         smoothing_length;
+                                         surface_normal_method=ColorfieldSurfaceNormal(),
+                                         reference_particle_spacing=particle_spacing)
+
+    if wall
+        boundary_system = create_boundary_system(coordinates, particle_spacing,
+                                                 state_equation, kernel, smoothing_length,
+                                                 NDIMS, walldistance)
+        semi = Semidiscretization(system, boundary_system)
+    else
+        semi = Semidiscretization(system)
+        boundary_system = nothing
+    end
+
+    ode = semidiscretize(semi, tspan)
+    TrixiParticles.update_systems_and_nhs(ode.u0.x..., semi, 0.0)
+
+    return system, boundary_system, semi, ode
+end
+
+function compute_and_test_surface_normals(system, semi, ode; NDIMS=2)
+    surface_tension = SurfaceTensionAkinci()
+
+    v0_ode, u0_ode = ode.u0.x
+    v = TrixiParticles.wrap_v(v0_ode, system, semi)
+    u = TrixiParticles.wrap_u(u0_ode, system, semi)
+
+    TrixiParticles.compute_surface_normal!(system, surface_tension, v, u, v0_ode, u0_ode,
+                                           semi, 0.0)
+
+    # After computation, check that surface normals have been computed and are not NaN or Inf
+    @test all(isfinite.(system.cache.surface_normal))
+    @test all(isfinite.(system.cache.neighbor_count))
+    @test size(system.cache.surface_normal, 1) == NDIMS
+
+    nparticles = size(u, 2)
+
+    # Check that the threshold has been applied correctly
+    threshold = 2^ndims(system) + 1
+
+    # # Test the surface normals based on neighbor counts
+    # Test that surface normals are zero when there are not enough neighbors.
+    # For the linear arrangement, surface normals may still be zero
+    # when we have more neighbors than the threshold.
+    for i in 1:nparticles
+        @test system.cache.neighbor_count[i] >= threshold ||
+              all(system.cache.surface_normal[:, i] .== 0.0)
+    end
```

Comment:
```suggestion
    @test all(i -> system.cache.neighbor_count[i] >= threshold || iszero(system.cache.surface_normal[:, i]), 1:nparticles)
```
This will generate only one test in the summary.